resources/frontend_shared/static_viz_interface.js

```diff
@@ -58,13 +58,14 @@ function categorical_donut(rows, colors, settings) {
   return StaticViz.RenderChart("categorical/donut", {
     data: toJSArray(rows),
     colors: toJSMap(colors),
-    settings: JSON.parse(settings)
+    settings: JSON.parse(settings),
   });
 }
 
-function progress(data, settings) {
+function progress(data, settings, colors) {
```

Comment:
the third argument `colors` is the instance colors passed from the setting `application-colors`